.github/workflows/dependabot-build.yml

```diff
@@ -25,6 +25,13 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v3
 
+      - name: Git config
+        run: |
+          git config --global user.email "github-actions@example.com"
+          git config --global user.name "github-actions[bot]"
+          git fetch origin ${{ github.head_ref }}
+          git checkout ${{ github.head_ref  }}
```

Comment:
@figi44 why do we need to fetch & checkout here when we already using `actions/checkout` above?

Reply:
good point, the same could be achieved with a `ref` input to `actions/checkout`

Reply:
Does `actions/checkout` not automatically checkout the correct ref?

Reply:
Had a chat about it - `actions/checkout` will checkout the internal GitHub merge commit for the PR, no the branch we need, so passing the `ref` in makes sense.